nuxeo-dm/nuxeo-platform-webapp-core/src/main/resources/OSGI-INF/theme-contrib.xml

```diff
@@ -405,167 +583,48 @@
-    <style name="print_generic">
-      <src>themes/css/print_generic.css</src>
-    </style>
-  </extension>
-
-  <!-- application settings -->
-  <extension target="org.nuxeo.theme.services.ThemeService" point="applications">
-
-    <application root="${org.nuxeo.ecm.contextPath}"
-      template-engine="jsf-facelets">
-
-      <negotiation>
-        <strategy>nuxeo5</strategy>
-        <default-engine>default</default-engine>
-        <default-theme>galaxy/default</default-theme>
-        <default-perspective>default</default-perspective>
-        <!-- TODO default collection/flavor... -->
-      </negotiation>
-
-      <!-- Cache control for theme resources (/nxthemes-lib/) -->
-      <resource-caching>
-        <lifetime>36000</lifetime>
-      </resource-caching>
-
-      <!-- Cache control for theme styles (/nxthemes-css/) -->
-      <style-caching>
-        <lifetime>900</lifetime>
-      </style-caching>
-
-      <view id="/create_relation_search_document_popup.xhtml">
-        <theme>galaxy/popup</theme>
-      </view>
-      <view id="/dndFormCollector.xhtml">
-        <theme>galaxy/popup</theme>
-      </view>
-      <view id="/editor_link_search_document.xhtml">
-        <theme>galaxy/popup</theme>
-      </view>
-      <view id="/editor_image_upload.xhtml">
-        <theme>galaxy/popup</theme>
-      </view>
-
-      <view id="/info_view.xhtml">
-        <theme>galaxy/popup</theme>
-      </view>
-
-    </application>
-  </extension>
-
-
-  <!-- negotiation strategy for nuxeo5 -->
-  <extension target="org.nuxeo.theme.services.ThemeService" point="negotiations">
+  <extension target="org.nuxeo.theme.styling.service" point="negotiations">
 
-    <negotiation object="theme" strategy="nuxeo5">
-      <scheme>org.nuxeo.theme.jsf.negotiation.theme.RequestParameter</scheme>
-      <scheme>org.nuxeo.theme.jsf.negotiation.theme.CookieValue</scheme>
-      <scheme>org.nuxeo.theme.jsf.negotiation.theme.ViewId</scheme>
-      <!-- local theme (specific to nuxeo5) -->
-      <scheme>org.nuxeo.ecm.localconf.LocalTheme</scheme>
-      <scheme>org.nuxeo.theme.jsf.negotiation.theme.DefaultTheme</scheme>
+    <negotiation target="jsfDefaultPage">
+      <negotiator class="org.nuxeo.ecm.web.resources.jsf.negotiators.DefaultPage"
+        order="100">
+        <property name="jsfDefaultPage">galaxy/default</property>
+      </negotiator>
     </negotiation>
 
-    <negotiation object="perspective" strategy="nuxeo5">
-      <scheme>
-        org.nuxeo.theme.jsf.negotiation.perspective.RequestParameter
-      </scheme>
-      <scheme>org.nuxeo.theme.jsf.negotiation.perspective.CookieValue</scheme>
-      <scheme>org.nuxeo.theme.jsf.negotiation.perspective.ViewId</scheme>
-      <!-- local perspective (specific to nuxeo5) -->
-      <scheme>
-        org.nuxeo.ecm.localconf.LocalPerspective
-      </scheme>
-      <scheme>
-        org.nuxeo.theme.jsf.negotiation.perspective.DefaultPerspective
-      </scheme>
+    <negotiation target="jsfPage">
```

Comment:
how about adding a `category="jsf"` like we do for layouts and widgets ?
something like `<negotiation target="page" category="jsf">`
then negotiators could use the `target` as the default value for the `param` property.